torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -469,6 +475,132 @@ std::string CudaCodeGen::GetUniqueFuncName(const std::string& func_prefix) {
   return func_prefix + "_" + std::to_string(value);
 }
 
+// Find all the statements that are not covered by any thread-idx axes,
+// and wrapp them under a trivial thread idx.
+class NoThreadIdxRewriter : public IRMutator {
+ private:
+  Stmt* rewrite(const std::vector<Stmt*>& stmts) {
+    std::vector<Stmt*> cloned_stmts(stmts.size());
+    for (int index = 0; index < stmts.size(); index++) {
+      cloned_stmts[index] = Stmt::clone(stmts[index]);
+    }
+    Stmt* new_block = Block::make(cloned_stmts);
+    // Wrap the new block under a trivial thread-idx
+    //   for t in 0..1: // threadIdx
+    //     if (t < 1):
+    //       new_block
+    VarHandle t("t", kInt);
+    ExprHandle t_lt_1 = CompareSelect::make(t, 1, CompareSelectOperation::kLT);
+    // TODO: move "if (t < 1)" to threadIdx generation
+    Cond* masked_block = Cond::make(t_lt_1, new_block, nullptr);
+    LoopOptions thread_idx_opt;
+    // TODO: the added trivial threadIdx needs to match the kernel threadIdx
+    // dimensions
+    thread_idx_opt.set_gpu_thread_index(0);
+    For* trivial_loop = For::make(t, 0, 1, masked_block, thread_idx_opt);
+    return trivial_loop;
+  }
+
+  Stmt* mutate(const For* v) override {
+    if (v->loop_options().is_gpu_block_index()) {
+      gpu_blocks_.push_back(v);
+      need_rewrite_ = false;
+    } else if (v->loop_options().is_gpu_thread_index()) {
+      gpu_threads_.push_back(v);
+    }
+
+    Stmt* new_for = IRMutator::mutate(v);
+
+    if (v->loop_options().is_gpu_block_index()) {
+      gpu_blocks_.pop_back();
+      need_rewrite_ = false;
+    } else if (v->loop_options().is_gpu_thread_index()) {
+      gpu_threads_.pop_back();
+    }
+
+    return new_for;
+  }
+
+  Stmt* mutate(const Block* v) override {
+    std::list<Stmt*> old_stmts = v->stmts();
+    std::vector<bool> need_rewrites(old_stmts.size());
+    std::vector<Stmt*> new_stmts(old_stmts.size());
+    ;
+    int index = 0;
+    for (auto old_stmt : old_stmts) {
+      need_rewrite_ = false;
+      Stmt* new_stmt = old_stmt->accept_mutator(this);
+      need_rewrites[index] = need_rewrite_;
+      new_stmts[index] = new_stmt;
+      index++;
+    }
+
+    bool any_need_fix = false;
+    bool all_need_fix = true;
+    for (auto need_fix : need_rewrites) {
+      if (need_fix) {
+        any_need_fix = true;
+      } else {
+        all_need_fix = false;
+      }
+    }
+
+    need_rewrite_ = false;
+    // If nothing needs fix, return as it is
+    if (!any_need_fix) {
+      return (Stmt*)v;
+    }
+
+    // If all needs fix, check if parent is a GPU block
+    if (all_need_fix) {
```

Comment:
Why do we need to special-case this? Won't the logic below handle it as well?

Reply:
Added a note to clarify.